### PR TITLE
fixed aggregator and tx processing

### LINKED ISSUE #37: Redo TxVerification
https://github.com/thehubbleproject/hubble-commander/blob/master/core/tx.go#L242

### PR DESCRIPTION
**Pull request by vaibhavchellani**
Fixes #37

### CODE REVIEW

File: aggregator/aggregator.go [hunk 1]
```diff
@@ -128,67 +129,54 @@ func (a *Aggregator) ProcessAndSubmitBatch(txs []core.Tx) {
...
-	// 		commitment := core.Commitment{Txs: txInCommitment, UpdatedRoot: updatedRoot, BatchType: tx.Type, AggregatedSignature: aggregatedSig.ToBytes()}
-	// 		commitments = append(commitments, commitment)
-	// 	}
-	// 	currentRoot = updatedRoot
-	// }
-	// elapsed := time.Since(start)
-	// log.Printf("Process batch took %s", elapsed)
-	// return commitments, nil
+	if len(txs) == 0 {
+		return commitments, errors.New("no tx to process,aborting")
```
Comment (ChihChengLiang) [inline aggregator/aggregator.go:133]: > errors.New("no tx to process,aborting")

Nitpick: Should we organize these errors to a `errors.go`?

File: aggregator/aggregator.go [hunk 2]
```diff
@@ -128,67 +129,54 @@ func (a *Aggregator) ProcessAndSubmitBatch(txs []core.Tx) {
...
+		rootAcc, err := a.DB.GetRoot()
+		if err != nil {
+			return commitments, err
+		}
+		a.Logger.Debug("Latest root", "root", rootAcc.Hash)
+		currentRoot, err := core.HexToByteArray(rootAcc.Hash)
+		if err != nil {
+			return commitments, err
+		}
+		fromAccProof, toAccProof, txDBConn, err := tx.GetVerificationData()
```
Comment (ChihChengLiang) [inline aggregator/aggregator.go:147]: We really should avoid the term "account" at all cost.
```suggestion
		fromStateProof, toStateProof, txDBConn, err := tx.GetVerificationData()
```

File: aggregator/aggregator.go [hunk 3]
```diff
@@ -128,67 +129,54 @@ func (a *Aggregator) ProcessAndSubmitBatch(txs []core.Tx) {
...
+			a.Logger.Error("Unable to create verification data", "error", err)
+			return commitments, err
+		}
+
+		newRoot, err := a.LoadedBazooka.ProcessTx(currentRoot, tx, fromAccProof, toAccProof)
+		if err != nil {
+			a.Logger.Error("Error processing tx", "tx", tx.String(), "error", err)
+			if txDBConn.Instance != nil {
+				txDBConn.Instance.Rollback()
+				txDBConn.Close()
```
Comment (ChihChengLiang) [inline aggregator/aggregator.go:158]: Nitpick: I remember golang has a `defer` feature, so if we defer closing DB connection at the start, we don't have to worry about forgetting it later.

File: aggregator/aggregator.go [hunk 4]
```diff
@@ -128,67 +129,54 @@ func (a *Aggregator) ProcessAndSubmitBatch(txs []core.Tx) {
...
+			}
+			return commitments, err
+		}
+		if txDBConn.Instance != nil {
+			txDBConn.Instance.Commit()
+			txDBConn.Close()
+		}
+
+		if i%32 == 0 {
```
Comment (ChihChengLiang) [inline aggregator/aggregator.go:167]: Nitpick: We should make this magic number 32 a global constant.

**Review by ChihChengLiang — APPROVED** [review]
Left some comments and nitpicks. Looks good once non-nitpicks issues are addressed

File: aggregator/aggregator.go (hunk 1 above)
Comment (vaibhavchellani) [reply aggregator/aggregator.go:133]: yeah we should have errors at compile time

File: aggregator/aggregator.go (hunk 2 above)
Comment (vaibhavchellani) [reply aggregator/aggregator.go:147]: yeah its deeply rooted right now, we will have to slowly remove it from the codebase

File: aggregator/aggregator.go (hunk 3 above)
Comment (vaibhavchellani) [reply aggregator/aggregator.go:158]: yep, `defer` is preferred here

File: aggregator/aggregator.go (hunk 4 above)
Comment (vaibhavchellani) [reply aggregator/aggregator.go:167]: haha yep for sure

Comment (vaibhavchellani) [reply aggregator/aggregator.go:167]: Will add it to config

**Pull request merged**